### ISSUE TITLE
Upgrade peerDependencies react 16.x.x

### ISSUE THREAD

**Issue by tphan18** [LouisBarranqueiro/reapop#57]
React just released 16.x.x, so we all have errors like this

```
error "reapop#react@^0.14.0 || ^15.0.0" doesn't satisfy found match of "react@16.0.0"
error "reapop#react-transition-group#react@^15.0.0" doesn't satisfy found match of "react@16.0.0"
error "reapop#react-transition-group#react-dom@^15.0.0" doesn't satisfy found match of "react-dom@16.0.0"
```

**Comment by human-a** [comment]
I am also getting the following error on react 16

```
Warning: Expected `onClick` listener to be a function, instead got a value of `string` type.
    in div (created by Notification)
    in Notification (created by Connect(Notification))
    in Connect(Notification) (created by NotificationsContainer)
    in CSSTransitionGroupChild (created by TransitionGroup)
    in span (created by TransitionGroup)
    in TransitionGroup (created by CSSTransitionGroup)
    in CSSTransitionGroup (created by NotificationsContainer)
    in div (created by NotificationsContainer)
    in NotificationsContainer (created by NotificationsSystem)
    in div (created by NotificationsSystem)
    in NotificationsSystem (created by Connect(NotificationsSystem))
    in Connect(NotificationsSystem) (at index.js:100)
    in div (created by FocusTrap)
    in FocusTrap (created by HotKeys)
    in HotKeys (at index.js:99)
    in Studio (created by Connect(Studio))
    in Connect(Studio) (created by RouterContext)
    in RouterContext (created by Router)
    in Router (at index.js:52)
    in Provider (at index.js:51)
```

**Comment by LouisBarranqueiro** [comment]
Hey @tphan18 @human-a , I released a fix in  [v1.1.4](https://github.com/LouisBarranqueiro/reapop/releases/tag/v1.1.4). I started working to officially support React 16. For that, I have to upgrade all the test setup to run all the tests suite with react 16, 15, 14.
